packages/core/src/components/region/RegionButton/RegionButton.tsx

```diff
@@ -1,27 +1,51 @@
-import { Button as UIButton } from '@faststore/ui'
+import { useEffect, useRef } from 'react'
 
-import { Icon, useUI } from '@faststore/ui'
+import { Button as UIButton, Icon as UIIcon, useUI } from '@faststore/ui'
+import { deliveryPromise, session as initialSession } from 'discovery.config'
 import { useSession } from 'src/sdk/session'
 import { textToTitleCase } from 'src/utils/utilities'
-import { session as initialSession } from 'discovery.config'
+
+import { useRegion } from '../RegionModal/useRegion'
 
 function RegionButton({ icon, label }: { icon: string; label: string }) {
-  const { openModal } = useUI()
+  const { openModal, openPopover } = useUI()
   const { city, postalCode } = useSession()
-  const shouldDisplayPostalCode = postalCode !== initialSession.postalCode
+  const { isValidationComplete } = useRegion()
+  const regionButtonRef = useRef<HTMLButtonElement>(null)
+
+  const defaultPostalCode =
+    !!initialSession?.postalCode && postalCode === initialSession.postalCode
+
+  // If location is not mandatory, and default zipCode is provided or if the user has not set a zipCode, show the popover.
+  const displayRegionPopover =
+    defaultPostalCode || (!postalCode && !deliveryPromise.mandatory)
+
+  useEffect(() => {
+    if (
+      isValidationComplete &&
+      displayRegionPopover &&
+      regionButtonRef.current
+    ) {
+      openPopover({
+        isOpen: true,
+        triggerRef: regionButtonRef,
+      })
+    }
+  }, [isValidationComplete, defaultPostalCode, openPopover])
 
   return (
-    <UIButton
-      variant="tertiary"
-      size="small"
-      icon={<Icon name={icon} width={18} height={18} weight="bold" />}
-      iconPosition="left"
-      onClick={openModal}
-    >
-      {city && postalCode
-        ? `${textToTitleCase(city)}${shouldDisplayPostalCode ? `, ${postalCode}` : ''}`
-        : label}
-    </UIButton>
+    <>
```

Comment:
question: is it necessary, or was it just forgotten to be removed?

Reply:
I've missed it! I'm removing it now! 👍